src/main/java/io/quarkus/search/app/indexing/FailureCollector.java

```diff
@@ -0,0 +1,165 @@
+            }
+        }
+    }
+
+    private static class GithubFailureReporter {
+
+        private static final String INDEX_STATUS_ISSUE_LABEL = "Autogenerated Indexing Status";
+        private static final String STATUS_CRITICAL = "Critical";
+        private static final String STATUS_WARNING = "Warning";
+        private static final String STATUS_SUCCESS = "Success";
+        private final GitHub github;
+        private final String repositoryName;
+        private final OptionalInt issueId;
+
+        GithubFailureReporter(String repositoryName, OptionalInt issueId) throws IOException {
+            this.github = GitHubBuilder.fromEnvironment().build();
+            this.repositoryName = repositoryName;
+            this.issueId = issueId;
+
+        }
+
+        void report(Map<Level, List<Failure>> failures) {
+            try {
+                GHRepository repository = github.getRepository(repositoryName);
+                GHIssue issue;
+                if (issueId.isPresent()) {
+                    issue = repository.getIssue(issueId.getAsInt());
+                } else {
+                    List<GHIssue> issues = repository.queryIssues().label(INDEX_STATUS_ISSUE_LABEL)
+                            .pageSize(1)
+                            .sort(GHIssueQueryBuilder.Sort.CREATED)
+                            .direction(GHDirection.DESC)
+                            .list().toList();
+                    if (!issues.isEmpty()) {
+                        issue = issues.get(0);
+                    } else {
+                        issue = repository.createIssue(issueTitle(""))
+                                .label(INDEX_STATUS_ISSUE_LABEL)
+                                .create();
+                    }
+                }
+
+                String status;
+                if (failures.get(Level.CRITICAL).isEmpty()) {
+                    if (failures.get(Level.WARNING).isEmpty()) {
+                        status = STATUS_SUCCESS;
+                    } else {
+                        status = STATUS_WARNING;
+                    }
+                } else {
+                    status = STATUS_CRITICAL;
+                }
+                issue.setTitle(issueTitle(status));
+
+                StringBuilder body = new StringBuilder("#### DO NOT EDIT THIS ISSUE MANUALLY!\n#### AUTOGENERATED ISSUE\n");
+                body.append("## search.quarkus.io indexing status report: ")
+                        .append(status).append('\n');
+                if (STATUS_SUCCESS.equals(status)) {
+                    body.append("Nothing to report.");
+                } else {
+                    for (Map.Entry<Level, List<Failure>> entry : failures.entrySet()) {
+                        report(body, entry.getValue(), entry.getKey());
+                    }
+                }
+
+                issue.setBody(body.toString());
+
+                if (!GHIssueState.OPEN.equals(issue.getState())) {
+                    issue.reopen();
```

Comment:
I think the idea of these reports is that:

1. We only comment when there's been a problem (to avoid pointless notifications)
2. We only reopen an issue when there's been a problem.

See https://github.com/quarkiverse/quarkiverse/issues/57
I'm not sure what controls this exactly, but I'm pretty sure it's open source, so you should be able to find the corresponding code somewhere in the `quarkiverse` org.